lib/PlacesProvider.js

```diff
@@ -158,28 +190,117 @@ Links.prototype = {
     // In general the groupby behavior in the absence of aggregates is not
     // defined in SQL, hence we are relying on sqlite implementation that may
     // change in the future.
-    let sqlQuery = "SELECT url, title, frecency, " +
-                   "       last_visit_date as lastVisitDate, favicon, mime_type, " +
-                   "       \"history\" as type " +
-                   "FROM " +
-                   "(    " +
-                   " SELECT rev_host, moz_places.url, moz_favicons.data as favicon, mime_type, title, frecency, last_visit_date " +
-                   " FROM moz_places " +
-                   " LEFT JOIN moz_favicons " +
-                   " ON favicon_id = moz_favicons.id " +
-                   " WHERE hidden = 0 AND last_visit_date NOTNULL " +
-                   " ORDER BY rev_host, frecency, last_visit_date, moz_places.url DESC " +
-                   ")    " +
-                   "GROUP BY rev_host " +
-                   "ORDER BY frecency DESC, lastVisitDate DESC, url " +
-                   "LIMIT :limit";
+    let sqlQuery = `SELECT url, title, frecency,
+                          last_visit_date as lastVisitDate, favicon, mime_type,
```

Comment:
ESLint has a [`camelcase`](http://eslint.org/docs/rules/camelcase.html) rule which causes 20 new errors in our code (yet none of them are related to this `mime_type` business).

Grep solves the day with this list, but I'm not sure what else out there may have odd underscores in the name that ESLint isn't catching:
```
✗ grep -irn "mime_type" . --exclude-dir=node_modules
./lib/PlacesProvider.js:162:                   "       last_visit_date as lastVisitDate, favicon, mime_type, " +
./lib/PlacesProvider.js:166:                   " SELECT rev_host, moz_places.url, moz_favicons.data as favicon, mime_type, title, frecency, last_visit_date " +
./lib/PlacesProvider.js:178:                  columns: ["url", "title", "lastVisitDate", "frecency", "favicon", "mime_type", "type"],
./lib/PlacesProvider.js:185:        link.favicon = `data:${link.mime_type};base64,${encodedData}`;
./lib/PlacesProvider.js:187:      delete link.mime_type;
```